File: app/src/main/java/com/esafirm/androidplayground/rxjava2/RxJava2SampleAct.java
package com.esafirm.androidplayground.rxjava2;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import com.esafirm.androidplayground.common.BaseAct;
import com.esafirm.androidplayground.common.MenuFactory;
import com.esafirm.androidplayground.common.OnNavigatePage;
import com.esafirm.androidplayground.utils.ActivityStater;

import java.util.Arrays;

public class RxJava2SampleAct extends BaseAct {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(
                MenuFactory.create(this, Arrays.asList(
                        "Observable vs Flowable",
                        "Interop with v1",
                        "Null item on Stream",
                        "Schedulers",
                        "RetryWhen",
                        "Single | Kotlin Infix Operator"
                ), new OnNavigatePage() {
                    @Override
                    public void navigate(int index) {
                        navigateToIndex(index);
                    }
                })
        );
    }

    private void navigateToIndex(int index) {
        switch (index) {
            case 0:
                ActivityStater.start(this, RxJava2ObservableFlowableAct.class);
                break;
            case 1:
                ActivityStater.start(this, RxJava2InteropAct.class);
                break;
            case 2:
                ActivityStater.start(this, RxJava2NullItemAct.class);
                break;
            case 3:
                ActivityStater.start(this, RxJava2SchedulerSampleAct.class);
                break;
            case 4:
                ActivityStater.start(this, RxJava2RetryWhen.class);
                break;
        }
    }

    /* --------------------------------------------------- */
    /* > Stater */
    /* --------------------------------------------------- */

    public static void start(Context context) {
        context.startActivity(
                new Intent(context, RxJava2SampleAct.class));
    }
}

File: app/src/main/java/com/esafirm/androidplayground/rxjava2/RxJava2SampleAct.kt
package com.esafirm.androidplayground.rxjava2

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.esafirm.androidplayground.common.BaseAct
import com.esafirm.androidplayground.common.MenuFactory
import com.esafirm.androidplayground.utils.ActivityStater

class RxJava2SampleAct : BaseAct() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(
            MenuFactory.create(this, listOf(
                "Observable vs Flowable",
                "Interop with v1",
                "Null item on Stream",
                "RetryWhen",
                "Single | Kotlin Infix Operator"
            )) {
                this.navigateToIndex(it)
            }
        )
    }

    private fun navigateToIndex(index: Int) {
        when (index) {
            0 -> ActivityStater.start(this, RxJava2ObservableFlowableAct::class.java)
            1 -> ActivityStater.start(this, RxJava2InteropAct::class.java)
            2 -> ActivityStater.start(this, RxJava2NullItemAct::class.java)
            3 -> ActivityStater.start(this, RxJava2RetryWhen::class.java)
        }
    }

    companion object {

        /* --------------------------------------------------- */
        /* > Stater */
        /* --------------------------------------------------- */

        fun start(context: Context) {
            context.startActivity(
                Intent(context, RxJava2SampleAct::class.java))
        }
    }
}
